### PR TITLE
Remove dependency on torchdata.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

Starting from torchdata 0.8, the datapipe related code will be removed, we temporarily move the dependency to torch.data, to see whether more deprecation will happen in torch.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/internal/datapipe_utils.py
Comment (mfbalin) [inline python/dgl/graphbolt/internal/datapipe_utils.py:1]: Why don't we import everything required for GraphBolt related to Datapipes in this file and import this file from the rest of GraphBolt? Then we could override any of these components one by one in the future with no change to the rest of the codebase.

Comment (mfbalin) [reply python/dgl/graphbolt/internal/datapipe_utils.py:1]: We could rename it to graphbolt/datapipes/utils and datapipes/datapipes.

Comment (frozenbugs) [reply python/dgl/graphbolt/internal/datapipe_utils.py:1]: Good point, I've done it for traverse_dps, for the other IterDataPipe / Mapper related, we can do it in the future in other PRs when needed.

**Comment by mfbalin** [comment]
@dgl-bot

File: tests/python/pytorch/graphbolt/test_item_sampler.py [hunk 1]
```diff
@@ -408,11 +408,10 @@ def test_append_with_other_datapipes():
     batch_size = 4
     item_set = gb.ItemSet(torch.arange(0, num_ids), names="seeds")
     data_pipe = gb.ItemSampler(item_set, batch_size)
-    # torchdata.datapipes.iter.Enumerator
-    data_pipe = data_pipe.enumerate()
-    for i, (idx, data) in enumerate(data_pipe):
-        assert i == idx
-        assert len(data.seeds) == batch_size
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_item_sampler.py:415]: This test case implicitly used datapipe, which break after we removed all datapipe dependency.

File: python/dgl/graphbolt/internal/datapipe_utils.py [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+
+__all__ = [
+    "datapipe_graph_to_adjlist",
+    "find_dps",
+    "replace_dp",
+    "traverse_dps",
+]
+
+
+PRODUCER_SLEEP_INTERVAL = 0.0001  # Interval between buffer fulfillment checks
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/internal/datapipe_utils.py:20]: where's these lines from? If literally copied, add the source as well?

File: python/dgl/graphbolt/internal/datapipe_utils.py [hunk 3]
```diff
@@ -51,3 +71,301 @@ def datapipe_graph_to_adjlist(datapipe_graph):
     result_dict = {}
     _get_parents(result_dict, datapipe_graph)
     return result_dict
+
+
+# Copied from:
+# https://github.com/pytorch/data/blob/88c8bdc6662f37649b7ea5df0bd90a4b24a56876/torchdata/dataloader2/graph/utils.py#L16
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/internal/datapipe_utils.py:77]: I prefer range: https://github.com/pytorch/data/blob/88c8bdc6662f37649b7ea5df0bd90a4b24a56876/torchdata/dataloader2/graph/utils.py#L16-L35

File: python/dgl/graphbolt/internal/datapipe_utils.py [hunk 4]
```diff
@@ -51,3 +71,301 @@ def datapipe_graph_to_adjlist(datapipe_graph):
...
+                dps.append(dp)
+            helper(src_graph)
+
+    helper(graph)
+
+    return dps
+
+
+# Copied from:
+# https://github.com/pytorch/data/blob/88c8bdc6662f37649b7ea5df0bd90a4b24a56876/torchdata/dataloader2/graph/utils.py#L82
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/internal/datapipe_utils.py:102]: ditto

**Review by Rhett-Ying — APPROVED** [review]
LGTM. please remember to remove other torchdata related code such as setup.py, dev.yaml.template and others after merging this PR.

**Pull request merged**